prolif/residue.py

```diff
@@ -152,20 +162,20 @@ class Residue(BaseRDKitMol):
     ``str(Residue)``
     """
 
-    def __init__(self, mol):
+    def __init__(self, mol: "Chem.Mol"):
         super().__init__(mol)
         FastFindRings(self)
         self.resid = ResidueId.from_atom(self.GetAtomWithIdx(0))
 
-    def __repr__(self):  # pragma: no cover
+    def __repr__(self) -> str:  # pragma: no cover
         name = ".".join([self.__class__.__module__, self.__class__.__name__])
         return f"<{name} {self.resid} at {id(self):#x}>"
 
-    def __str__(self):
+    def __str__(self) -> str:
         return str(self.resid)
 
 
-class ResidueGroup(UserDict):
+class ResidueGroup(UserDict["ResidueId", "Residue"]):
```

Comment:
In terms of style for the annotations, I actually prefer not to use the string-literal version, with the exception of a class referencing itself, referencing a class defined later in the file, or imports that aren't required at runtime. Could you make changes accordingly throughout this file please?